### DISCUSSION TITLE
renvoate Failed git push error

### DISCUSSION THREAD

**Discussion by Rahul-learn** [renovatebot/renovate#10314] [Request Help]
I am seeing below error in renovate logs. Could you please help?

  \nerror: failed to push some refs to 'https://github.****.com/****-org/****service.git'\n\n    at GitExecutorChain.onFatalException (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:67:77)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:59:21)\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"

**Comment by rarkins** [comment]
Capture the debug logs before and after it happens.

**Comment by Rahul-learn** [comment]
@rarkins 

Debug logs

```
DEBUG: Committing files to branch renovate/com.google.code.gson-gson-2.x (repository=*****/******, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: Error committing files (repository=***/****, branch=renovate/com.google.code.gson-gson-2.x)

DEBUG: Error committing files (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
       "err": {
         "task": {
           "commands": [
             "push",
             "origin",
             "renovate/com.google.code.gson-gson-2.x:renovate/com.google.code.gson-gson-2.x",
             "--force",
             "-u",
             "--no-verify",
             "--verbose",
             "--porcelain"
           ],
           "format": "utf-8"
         }

DEBUG: Passing repository-changed error up (repository=*******/*******, branch=renovate/com.google.code.gson-gson-2.x)
```

**Reply by rarkins** [reply]
Didn't you previously mention that you had forbidden force push on your server?

**Comment by Rahul-learn** [comment]
Yes, but here I am seeing different error not related to force push. Do you think its related to force push ?

**Reply by rarkins** [reply]
According to the log it's a git push with --force flag.

**Reply by Rahul-learn** [reply]
@rarkins 

Have enabled force push still we are seeing the same error. 

We are seeing this error only for this repository. We are getting PR's for other repos.


\nerror: failed to push some refs to 'https://github.****.com/****-org/****service.git'\n\n at GitExecutorChain.onFatalException (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:67:77)\n at GitExecutorChain. (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:59:21)\n at Generator.throw ()\n at rejected (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n at runMicrotasks ()\n at processTicksAndRejections (internal/process/task_queues.js:93:5)"

**Reply by viceice** [reply]
Please provide more context debug log lines.

**Reply by Rahul-learn** [reply]
@viceice @rarkins 

Here is the debug logs

DEBUG: gradle.updateDependency(): packageFile:settings.gradle depName:com.google.code.gson:gson, version:2.8.1 ==> 2.8.7 (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: gradle.updateDependency(): packageFile:liquibase.gradle depName:com.google.code.gson:gson, version:2.8.1 ==> 2.8.7 (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: gradle.updateDependency(): packageFile:gradle/sonarqube.gradle depName:com.google.code.gson:gson, version:2.8.1 ==> 2.8.7 (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: gradle.updateDependency(): packageFile:gradle.properties depName:com.google.code.gson:gson, version:2.8.1 ==> 2.8.7 (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: gradle.updateDependency(): packageFile:build.gradle depName:com.google.code.gson:gson, version:2.8.1 ==> 2.8.7 (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: Updating packageFile content (repository=****/****, packageFile=build.gradle, branch=renovate/com.google.code.gson-gson-2.x)
       "depName": "com.google.code.gson:gson"
DEBUG: Updated 1 package files (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: No updated lock files in branch (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: 1 file(s) to commit (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: Committing files to branch renovate/com.google.code.gson-gson-2.x (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
DEBUG: Error committing files (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
       "err": {
         "task": {
           "commands": [
             "push",
             "origin",
             "renovate/com.google.code.gson-gson-2.x:renovate/com.google.code.gson-gson-2.x",
             "--force",
             "-u",
             "--no-verify",
             "--verbose",
             "--porcelain"
           ],
           "format": "utf-8"
         },
         "message": "Pushing to https://github.********.com/****/****.git\nPOST git-receive-pack (621 bytes)\nremote:   ___ ___ ___   _ _____ ___ ___          \nremote:  / __| __|   \\ / \\_   _| __|   \\ (R)        \nremote:  \\__ \\ _|| |) / A \\| | | _|| |) |        \nremote:  |___/___|___/_/ \\_\\_| |___|___/         \nremote:  https://github.********.com/        \nremote: \nremote: Repo Name = ****/****        \nremote: ========================================        \nremote: ***Commits scanned (1 total):        \nremote: ******       \nremote: ========================================        \nremote: Violations listed, by line, below (results may be obfuscated):        \nremote: NONE        \nremote:       .--._____.--.              \nremote:       ( /_.) (._\\ )              \nremote:   __   |  / o \\  |   __          \nremote:  (oo\\   \\ \\_-_/ /   /oo)         \nremote:   \\  \\__/       \\__/  /          \nremote:    \\   /         \\   /           \nremote: Good job! No sensitive data found in your commit(s)!        \nremote: Push ACCEPTED        \nremote: ========================================        \nremote: 44 ms        \nerror: failed to push some refs to 'https://github.********.com/****/****.git'\n",
         "stack": "Error: Pushing to https://github.********.com/****/****.git\nPOST git-receive-pack (621 bytes)\nremote:   ___ ___ ___   _ _____ ___ ___          \nremote:  / __| __|   \\ / \\_   _| __|   \\ (R)        \nremote:  \\__ \\ _|| |) / A \\| | | _|| |) |        \nremote:  |___/___|___/_/ \\_\\_| |___|___/         \nremote:  https://github.********.com/        \nremote: \nremote: Repo Name = ****/****        \nremote: ========================================        \nremote: ***Commits scanned (1 total):        \nremote: ****        \nremote: ========================================        \nremote: Violations listed, by line, below (results may be obfuscated):        \nremote: NONE        \nremote:       .--._____.--.              \nremote:       ( /_.) (._\\ )              \nremote:   __   |  / o \\  |   __          \nremote:  (oo\\   \\ \\_-_/ /   /oo)         \nremote:   \\  \\__/       \\__/  /          \nremote:    \\   /         \\   /           \nremote: Good job! No sensitive data found in your commit(s)!        \nremote: Push ACCEPTED        \nremote: ========================================        \nremote: 44 ms        \nerror: failed to push some refs to 'https://github.********.com/****/****.git'\n\n    at GitExecutorChain.onFatalException (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:67:77)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:59:21)\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)"
       }
DEBUG: Passing repository-changed error up (repository=****/****, branch=renovate/com.google.code.gson-gson-2.x)
 INFO: Repository has changed during renovation - aborting (repository=****/****)
DEBUG: Repository timing splits (milliseconds) (repository=****/****)
       "splits": {"init": 2125, "extract": 145837, "lookup": 19120},
       "total": 171446

**Reply by rarkins** [reply]
Unfortunately it doesn't give a hint as to why it's being rejected:

```
Pushing to https://github.********.com/****/****.git
POST git-receive-pack (621 bytes)
remote: ___ ___ ___ _ _____ ___ ___ 
remote: / | | \ / \ | | \ (R) 
remote: \ \ || |) / A \| | | || |) | 
remote: |/|// \\| ||/ 
remote: https://github.********.com/ 
remote: 
remote: Repo Name = / 
remote: ======================================== 
remote: Commits scanned (1 total): 
remote: ****** 
remote: ======================================== 
remote: Violations listed, by line, below (results may be obfuscated): 
remote: NONE 
remote: .--..--. 
remote: ( /.) (.\ ) 
remote: __ | / o \ | __ 
remote: (oo\ \ \-/ / /oo) 
remote: \ \/ \/ / 
remote: \ / \ / 
remote: Good job! No sensitive data found in your commit(s)! 
remote: Push ACCEPTED 
remote: ======================================== 
remote: 44 ms 
error: failed to push some refs to 'https://github.********.com/****/****.git'
```

Does the branch already exist in the repo on the server side?

**Reply by viceice** [reply]
We saw done time ago that a branch named like the renovate branch prefix will block pushes. So please check if there is any remote branche named like `renovate`
